api/types.go

```diff
@@ -0,0 +1,141 @@
+package api
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+)
+
+// LoadBalancerList is a collection of LoadBalancer
+type LoadBalancerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LoadBalancer `json:"items"`
+}
+
+// +genclient=true
+
+// LoadBalancer describe a LoadBalancer witch providers Load balancing for services
+type LoadBalancer struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the LoadBalancer
+	LoadBalancerSpec `json:"spec,omitempty"`
+}
+
+// LoadBalancerSpec is a description of a LoadBalancer
+type LoadBalancerSpec struct {
+	// Type determines the type of LoadBalancer
```

Comment:
You should also mention the valid options, e.g.

Type determines the type of LoadBalancer, valid options are: external, internal.  For internal type, the specifications for providers can ONLY be service.